src/widgetresize.js

```diff
@@ -95,15 +112,58 @@ export default class WidgetResize extends Plugin {
 
 		resizer.attach();
 
-		this.editor.editing.view.once( 'render', () => resizer.redraw() );
-
-		this.resizers.push( resizer );
+		this.resizers.set( options.viewElement, resizer );
 
 		return resizer;
 	}
 
+	/**
+	 * Adds a listener that keep the track of currently focused resizer.
+	 *
+	 * @private
+	 */
+	_attachFocusChangeListener() {
+		this.editor.editing.downcastDispatcher.on( 'selection', ( evt, data, conversionApi ) => {
```

Comment:
I think that it's enough to:

1. Listen to `editing.view.selection#change`
2. Use `selection.getSelectedElement()` because that's the only time when the resizers are visible
3. After retrieving the element, use some `this.getResizerByHost()` method (which needs to be introduced as commented in other place). No other checks seem to be needed.

Reply:
That's right it was possible to simplify it, thanks!